File: backend/fief/schemas/oauth_account.py
from pydantic import UUID4

from fief.schemas.generics import CreatedUpdatedAt, UUIDSchema
from fief.schemas.oauth_provider import OAuthProvider


class OAuthAccount(UUIDSchema, CreatedUpdatedAt):
    account_id: str
    oauth_provider_id: UUID4
    oauth_provider: OAuthProvider

File: admin-frontend/src/schemas/oauth-account.ts
import { CreatedUpdatedAt, UUIDSchema } from './generics';
import { OAuthProvider } from './oauth-provider';

export interface OAuthAccount extends UUIDSchema, CreatedUpdatedAt {
  account_id: string;
  oauth_provider_id: string;
  oauth_provider: OAuthProvider;
}
